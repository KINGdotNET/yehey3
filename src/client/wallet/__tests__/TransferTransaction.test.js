import React from 'react';
import { shallow } from 'enzyme';
import TransferTransaction from '../TransferTransaction';

describe('(Component) TransferTransaction', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        to: 'helloezira',
        memo: 'Test Transfer Transaction',
        amount: <span>{'0 ECO'}</span>,
        timestamp: '0',
      };
      const wrapper = shallow(<TransferTransaction {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
