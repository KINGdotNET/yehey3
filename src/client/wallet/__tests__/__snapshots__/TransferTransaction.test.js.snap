// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TransferTransaction with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransferTransaction
    amount={
      <span>
        0 TME
      </span>
    }
    memo="Test Transfer Transaction"
    timestamp="0"
    to="helloWeYouMe"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserWalletTransactions__avatar"
        >
          <Avatar
            size={40}
            username="helloWeYouMe"
          />
        </div>,
        <div
          className="UserWalletTransactions__content"
        >
          <div
            className="UserWalletTransactions__content-recipient"
          >
            <FormattedMessage
              defaultMessage="Transferred to {username}"
              id="transferred_to"
              values={
                Object {
                  "username": <Link
                    replace={false}
                    to="/@helloWeYouMe"
                  >
                    <span
                      className="username"
                    >
                      helloWeYouMe
                    </span>
                  </Link>,
                }
              }
            />
            <span
              className="UserWalletTransactions__transfer"
            >
              - 
              <span>
                0 TME
              </span>
            </span>
          </div>
          <span
            className="UserWalletTransactions__timestamp"
          >
            <BTooltip
              title={
                <span>
                  <FormattedDate
                    value="0Z"
                  />
                   
                  <FormattedTime
                    value="0Z"
                  />
                </span>
              }
            >
              <span>
                <FormattedRelative
                  updateInterval={10000}
                  value="0Z"
                />
              </span>
            </BTooltip>
          </span>
          <span
            className="UserWalletTransactions__memo"
          >
            Test Transfer Transaction
          </span>
        </div>,
      ],
      "className": "UserWalletTransactions__transaction",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Avatar
            size={40}
            username="helloWeYouMe"
          />,
          "className": "UserWalletTransactions__avatar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 40,
            "username": "helloWeYouMe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Transferred to {username}"
                id="transferred_to"
                values={
                  Object {
                    "username": <Link
                      replace={false}
                      to="/@helloWeYouMe"
                    >
                      <span
                        className="username"
                      >
                        helloWeYouMe
                      </span>
                    </Link>,
                  }
                }
              />
              <span
                className="UserWalletTransactions__transfer"
              >
                - 
                <span>
                  0 TME
                </span>
              </span>
            </div>,
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>,
            <span
              className="UserWalletTransactions__memo"
            >
              Test Transfer Transaction
            </span>,
          ],
          "className": "UserWalletTransactions__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormattedMessage
                  defaultMessage="Transferred to {username}"
                  id="transferred_to"
                  values={
                    Object {
                      "username": <Link
                        replace={false}
                        to="/@helloWeYouMe"
                      >
                        <span
                          className="username"
                        >
                          helloWeYouMe
                        </span>
                      </Link>,
                    }
                  }
                />,
                <span
                  className="UserWalletTransactions__transfer"
                >
                  - 
                  <span>
                    0 TME
                  </span>
                </span>,
              ],
              "className": "UserWalletTransactions__content-recipient",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "Transferred to {username}",
                  "id": "transferred_to",
                  "values": Object {
                    "username": <Link
                      replace={false}
                      to="/@helloWeYouMe"
                    >
                      <span
                        className="username"
                      >
                        helloWeYouMe
                      </span>
                    </Link>,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "- ",
                    <span>
                      0 TME
                    </span>,
                  ],
                  "className": "UserWalletTransactions__transfer",
                },
                "ref": null,
                "rendered": Array [
                  "- ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "0 TME",
                    },
                    "ref": null,
                    "rendered": "0 TME",
                    "type": "span",
                  },
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>,
              "className": "UserWalletTransactions__timestamp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>,
                "title": <span>
                  <FormattedDate
                    value="0Z"
                  />
                   
                  <FormattedTime
                    value="0Z"
                  />
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "updateInterval": 10000,
                    "value": "0Z",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Transfer Transaction",
              "className": "UserWalletTransactions__memo",
            },
            "ref": null,
            "rendered": "Test Transfer Transaction",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserWalletTransactions__avatar"
          >
            <Avatar
              size={40}
              username="helloWeYouMe"
            />
          </div>,
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Transferred to {username}"
                id="transferred_to"
                values={
                  Object {
                    "username": <Link
                      replace={false}
                      to="/@helloWeYouMe"
                    >
                      <span
                        className="username"
                      >
                        helloWeYouMe
                      </span>
                    </Link>,
                  }
                }
              />
              <span
                className="UserWalletTransactions__transfer"
              >
                - 
                <span>
                  0 TME
                </span>
              </span>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>
            <span
              className="UserWalletTransactions__memo"
            >
              Test Transfer Transaction
            </span>
          </div>,
        ],
        "className": "UserWalletTransactions__transaction",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Avatar
              size={40}
              username="helloWeYouMe"
            />,
            "className": "UserWalletTransactions__avatar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 40,
              "username": "helloWeYouMe",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="UserWalletTransactions__content-recipient"
              >
                <FormattedMessage
                  defaultMessage="Transferred to {username}"
                  id="transferred_to"
                  values={
                    Object {
                      "username": <Link
                        replace={false}
                        to="/@helloWeYouMe"
                      >
                        <span
                          className="username"
                        >
                          helloWeYouMe
                        </span>
                      </Link>,
                    }
                  }
                />
                <span
                  className="UserWalletTransactions__transfer"
                >
                  - 
                  <span>
                    0 TME
                  </span>
                </span>
              </div>,
              <span
                className="UserWalletTransactions__timestamp"
              >
                <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />
                       
                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>
              </span>,
              <span
                className="UserWalletTransactions__memo"
              >
                Test Transfer Transaction
              </span>,
            ],
            "className": "UserWalletTransactions__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FormattedMessage
                    defaultMessage="Transferred to {username}"
                    id="transferred_to"
                    values={
                      Object {
                        "username": <Link
                          replace={false}
                          to="/@helloWeYouMe"
                        >
                          <span
                            className="username"
                          >
                            helloWeYouMe
                          </span>
                        </Link>,
                      }
                    }
                  />,
                  <span
                    className="UserWalletTransactions__transfer"
                  >
                    - 
                    <span>
                      0 TME
                    </span>
                  </span>,
                ],
                "className": "UserWalletTransactions__content-recipient",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "defaultMessage": "Transferred to {username}",
                    "id": "transferred_to",
                    "values": Object {
                      "username": <Link
                        replace={false}
                        to="/@helloWeYouMe"
                      >
                        <span
                          className="username"
                        >
                          helloWeYouMe
                        </span>
                      </Link>,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "- ",
                      <span>
                        0 TME
                      </span>,
                    ],
                    "className": "UserWalletTransactions__transfer",
                  },
                  "ref": null,
                  "rendered": Array [
                    "- ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "0 TME",
                      },
                      "ref": null,
                      "rendered": "0 TME",
                      "type": "span",
                    },
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />
                       
                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>,
                "className": "UserWalletTransactions__timestamp",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>,
                  "title": <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "updateInterval": 10000,
                      "value": "0Z",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Transfer Transaction",
                "className": "UserWalletTransactions__memo",
              },
              "ref": null,
              "rendered": "Test Transfer Transaction",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
